front/app/modules/commercial/custom_topics/admin/components/CustomTopicRow.tsx

```diff
@@ -1,68 +1,115 @@
 import React, { memo } from 'react';
-import { ITopicData } from 'services/topics';
-import { isNilOrError } from 'utils/helperUtils';
-import styled from 'styled-components';
+import { Row } from 'components/admin/ResourceList';
 
 // components
-import { Row } from 'components/admin/ResourceList';
 import Button from 'components/UI/Button';
 import { RowContent, RowContentInner, RowTitle } from './RowStyles';
+import { IconTooltip, colors } from '@citizenlab/cl2-component-library';
+import Link from 'utils/cl-router/Link';
+
+// resources
+import useCustomPages from 'hooks/useCustomPages';
+
+// hooks
+import useLocalize from 'hooks/useLocalize';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import styled from 'styled-components';
 
 // i18n
 import messages from './messages';
 import { FormattedMessage } from 'utils/cl-intl';
 
+// types
+import { ITopicData } from 'services/topics';
+
 const Buttons = styled.div`
   display: flex;
   align-items: center;
 `;
 
+export const StyledLink = styled(Link)`
+  color: ${colors.white} !important;
+  text-decoration: underline;
+
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
 interface Props {
-  topic: ITopicData | Error;
+  topic: ITopicData;
   isLastItem: boolean;
   handleDeleteClick: (topicId: string) => (event: React.FormEvent<any>) => void;
 }
 
 const CustomTopicRow = memo((props: Props) => {
   const { isLastItem, topic, handleDeleteClick } = props;
+  const localize = useLocalize();
 
-  if (!isNilOrError(topic)) {
-    return (
-      <Row
-        key={topic.id}
-        id={topic.id}
-        className="e2e-topic-field-row"
-        isLastItem={isLastItem}
-      >
-        <RowContent>
-          <RowContentInner>
-            <RowTitle value={topic.attributes.title_multiloc} />
-          </RowContentInner>
-        </RowContent>
-        <Buttons>
-          <Button
-            linkTo={`/admin/settings/topics/${topic.id}/edit`}
-            buttonStyle="secondary"
-            icon="edit"
-            id="e2e-custom-topic-edit-button"
-          >
-            <FormattedMessage {...messages.editButtonLabel} />
-          </Button>
+  const static_page_ids = topic.attributes.static_page_ids;
```

Comment:
I suppose we can use `topic.relationships.static_pages` here. The actual static page records will be cached (because we include them with `has_many` in the serializer on the BE), and so we can call `useCustomPage({ customPageId: id })` instead of `useCustomPages({ ids: static_page_ids })` and it will not send any request to the BE.

As an alternative, we indeed can use `static_pages_ids`. In this case, we don't need `has_many` in the serializer. But it probably doesn't follow [JSON API spec](https://jsonapi.org/format/).

Reply:
I changed it to use relationships, there are a few examples of doing that throughout the app, and I think with the caching system it's better to use that. 

I think the caching still works with `useCustomPages` instead of the singular one, if I go to areas/topics the only call I see is for that entity with the `included` query param, and the feature still works. So I think the relationship/caching works as intended. In this case I think we want `useCustomPages` too because we want to display a list of the linked ones, not show them one at a time

Reply:
Ah it looks like `useCustomPages` uses the same `customPageByIdStream(id)` as `useCustomPage`. Cool then.